File: forhksong/A.題目/python/A1-版本轉換/A1-1.py
##====================================
##題目說明：f-string (Python 3.6+ 支援)
##問題：使用 f-string 格式化字串，但 f-string 在 Python 3.6 之前不被支援。
##
##題目要求：請透過LLM及相關tooling，調整程式，使得可以在Java17 環境中運行。
##
##驗證：
##請確認參賽包workshop_tool腳本已運行，其中java-17-app為已準備好的java 17 SDK runtime container，可執行下列指令獲得編譯結果
##
##>> docker run --rm -v $(pwd):/app java-17-app
##>> 會出現Exception in thread "main" java.lang.reflect.InaccessibleObjectException

##內部參考：在 Java 7 中，反射可以輕鬆訪問私有字段，但在 Java 17 的模塊化系統中，這種行為受到了更多限制，會導致 IllegalAccessException。
##====================================

def greet(name):
    print(f"Hello, {name}!")  # 錯誤：舊版本不支援 f-string

greet("Alice")


#可能的錯誤訊息： SyntaxError: invalid syntax

File: forhksong/A.題目/java/A1-版本轉換/A1-1.java
//========================================================================
//題目說明：以下程式只能在較舊版本的SDK運行，請透過自動化工具調整為在較新的SDK也可運行
//
//提示(題目組內部參考)：以下程式用了一個特殊的類，這個類在 Java 9 中是允許的行為，但在 Java 17 中已被移除或更改，從而導致在 Java 17 中運行會出錯。
//
//題目Input： ./java/A1-1.java
//
//題目Output： 生成的檔案需為 ./java/A1-1-answer.java
//
//開發環境：
//請確認參賽包workshop_tool腳本已運行，將產生名為java-A1-1 container，可自動將./java/A1-1-answer.java程式在Java 17環境編譯並執行
//
//指令： docker run --rm -v ./java:/host-java-files java-A1-1 
//
//驗證： 校驗腳本將分別對A1-1.java、A1-1-answer.java 兩個程式輸出結果比對，確認.1)編譯結果正確及.2)程式輸出結果一致。
//========================================================================



import javax.security.cert.X509Certificate;

public class A1-1 {
    public static void main(String[] args) {
        try {
            X509Certificate cert = X509Certificate.getInstance(new byte[0]);
            System.out.println("Certificate: " + cert);
        } catch (Exception e) {
            e.printStackTrace();
        }
    }
}
